AsyncDisplayKit/Details/_ASDisplayView.mm

```diff
@@ -375,3 +378,126 @@ - (UIView *)preferredFocusedView
+
+@implementation UIAccessibilityElement (_ASDisplayView)
+
+- (void)setAsyncdisplaykit_node:(ASDisplayNode *)node
+{
+  objc_setAssociatedObject(self, ASDisplayNodeAssociatedNodeKey, node, OBJC_ASSOCIATION_ASSIGN); // Weak reference to avoid cycle, since the node retains the layer.
+  
+  // Update UIAccessibilityElement properties from node
+  self.accessibilityIdentifier = node.accessibilityIdentifier;
+  self.accessibilityLabel = node.accessibilityLabel;
+  self.accessibilityHint = node.accessibilityHint;
+  self.accessibilityValue = node.accessibilityValue;
+  self.accessibilityTraits = node.accessibilityTraits;
+}
+
+- (ASDisplayNode *)asyncdisplaykit_node
+{
+  return objc_getAssociatedObject(self, ASDisplayNodeAssociatedNodeKey);
+}
+
+@end
+
+@implementation _ASDisplayView (UIAccessibilityContainer)
+
+#pragma mark - UIAccessibility
+
+- (NSArray *)accessibleElements
+{
+  if ( _accessibleElements != nil ) {
+    return _accessibleElements;
+  }
+  
+  _accessibleElements = [[NSMutableArray alloc] init];
+ 
+  ASDisplayNode *selfNode = self.asyncdisplaykit_node;
+  
+  // Handle rasterize case
+  if (selfNode.shouldRasterizeDescendants) {
+    // In this case we have to go through the whole subnodes tree in BFS fashion and create all
+    // accessibility elements ourselves as the view hierarchy is flattened
+    
+    // Queue used to keep track of subnodes while traversing this layout in a BFS fashion.
+    std::queue<ASDisplayNode *> queue;
+    queue.push(selfNode);
+    
+    while (!queue.empty()) {
+      ASDisplayNode *node = queue.front();
+      queue.pop();
+      
+      // Check if we have to add the node to the accessiblity nodes as it's an accessiblity element
+      if (node != selfNode && node.isAccessibilityElement) {
+        UIAccessibilityElement *accessibilityElement = [[UIAccessibilityElement alloc] initWithAccessibilityContainer:self];
+        accessibilityElement.asyncdisplaykit_node = node;
+        [_accessibleElements addObject:accessibilityElement];
+      }
+
+      // Add all subnodes to process in next step
+      for (int i = 0; i < node.subnodes.count; i++)
+        queue.push(node.subnodes[i]);
+    }
+    return _accessibleElements;
+  }
+  
+  // Handle not rasterize case
+  // Create UI accessiblity elements for each subnode that represent an elment within the accessibility container
+  for (ASDisplayNode *subnode in selfNode.subnodes) {
+      // Check if this subnode is a UIAccessibilityContainer
+    if (!subnode.isAccessibilityElement && [subnode accessibilityElementCount] > 0) {
+      // We are good and the view is an UIAccessibilityContainer so add that
+      [_accessibleElements addObject:subnode.view];
+    } else if (subnode.isAccessibilityElement) {
+      // Create a accessiblity element from the subnode
+      UIAccessibilityElement *accessibilityElement = [[UIAccessibilityElement alloc] initWithAccessibilityContainer:self];
```

Comment:
I'm not sure I understand the need for this - can't the ASDisplayNode objects implement the UIAccessibilityElement protocol, as they already have most of the properties?  This would also avoid object creation, but more importantly, the additional code to support associated objects etc.